src/commands/list.rs

```diff
@@ -28,13 +29,25 @@ impl ListCommand {
 
                 // TODO: better error handling for writeln!
                 match entity.unwrap_or(Entity::TimeEntry) {
-                    Entity::TimeEntry => entities
-                        .time_entries
-                        .iter()
-                        .take(count.unwrap_or(usize::MAX))
-                        .for_each(|time_entry| {
-                            writeln!(handle, "{time_entry}").expect("failed to print")
-                        }),
+                    Entity::TimeEntry => {
+                        if json.unwrap_or(false) {
+                            let entries: Vec<_> = entities
+                                .time_entries
+                                .iter()
+                                .take(count.unwrap_or(usize::MAX))
+                                .collect();
+                            let json_string = serde_json::to_string_pretty(&entries).unwrap();
```

Comment:
```suggestion
                            let json_string = serde_json::to_string_pretty(&entries).expect("failed to serialize time entries to JSON");
```

I'd throw in an `expect` in place of the `unwrap`

Reply:
Good suggestion. Thanks